src/clarity-angular/datagrid/render/main-renderer.ts

```diff
@@ -3,21 +3,37 @@
  * This software is released under MIT license.
  * The full license information can be found in LICENSE in the root directory of this project.
  */
-import {Directive, ContentChildren, QueryList, AfterContentInit, AfterViewInit, OnDestroy} from "@angular/core";
+import {
+    Directive, ContentChildren, QueryList, AfterContentInit, AfterViewInit, OnDestroy,
+    ElementRef, Renderer
+} from "@angular/core";
 import {Subscription} from "rxjs/Subscription";
 import {DomAdapter} from "./dom-adapter";
 import {DatagridRenderOrganizer} from "./render-organizer";
 import {DatagridHeaderRenderer} from "./header-renderer";
 import {Items} from "../providers/items";
+import {Page} from "../providers/page";
 
 @Directive({
     selector: "clr-datagrid",
     providers: [DomAdapter]
 })
 export class DatagridMainRenderer implements AfterContentInit, AfterViewInit, OnDestroy {
 
-    constructor(private organizer: DatagridRenderOrganizer, private items: Items) {
-        this._subscriptions.push(organizer.computeWidths.subscribe(() => this.computeHeadersWidth()));
+    constructor(
+        private organizer: DatagridRenderOrganizer,
+        private items: Items,
+        private page: Page,
+        private domAdapter: DomAdapter,
+        private el: ElementRef,
+        private renderer: Renderer) {
+            this._subscriptions.push(organizer.computeWidths.subscribe(() => this.computeHeadersWidth()));
+            this._subscriptions.push(organizer.done.subscribe(() => {
+                if (!this._heightSet && this.page.size > 0) {
+                    this.computeDatagridHeight();
+                }
+                this._heightSet = true;
+            }));
```

Comment:
I anticipate one issue here: a paginated datagrid where the data arrives asynchronously. You should test this with a `setTimeout()` in a demo. I believe you will end up setting a height that is the one of an empty datagrid.

I think what you want is to compute the height when the page size is set _and_ the items are here.